clang/test/Driver/clang-sycl-linker-test.cpp

```diff
@@ -1,48 +1,35 @@
 // Tests the clang-sycl-linker tool.
 //
 // Test a simple case without arguments.
-// RUN: %clangxx -emit-llvm -c %s -o %t_1.bc
-// RUN: %clangxx -emit-llvm -c %s -o %t_2.bc
-// RUN: clang-sycl-linker --dry-run -triple spirv64 %t_1.bc %t_2.bc -o a.spv 2>&1 \
+// RUN: %clangxx -c -target spirv64 %s -o %t_1.o
+// RUN: %clangxx -c -target spirv64 %s -o %t_2.o
+// RUN: clang-sycl-linker --dry-run %t_1.o %t_2.o -o a.spv 2>&1 \
```

Comment:
We do not need to pass -triple to clang-sycl-linker. We will get  that information from the device image. Also, the clang-sycl-linker can now accept .o fat object files (instead of .bc files). This is more aligned with the community changes in clang-linker-wrapper

Reply:
> Doesn't `%clangxx -c -target spirv64 %s -o %t_1.o` create a SPIRV bytecode, not an (ELF) fat object?

I think @jzc implies that the output file extensions should be `.spv` not `.o`.

Reply:
'clang++ -c' compilation will result in a .o (fat binary) file, not a .spv file.

Thanks

Reply:
It will not unless you manually set the offloading model.
Your command should produce SPIR-V file.

Reply:
Hmm...Yes...you are right. It does create a SPIR-V file, albeit with a .o extension. Let me revisit this test.
Thanks

Reply:
Since these tests are all '--dry-run' tests (except the 'Unsupported file type' case, where I manually use packager to package the file), we do not reach the stage where we actually look inside the objects to see what it has.

However, it is too early to add such tests upstream. I have locally tested my changes with bitcodes and fat objects created using our downstream SYCL compiler.

I think I might as well just create dummy bitcode and object files here.

Thanks

Reply:
After some discussions, I have added test to verify the linked module.

Thanks